app/services/update_type_service.rb

```diff
@@ -35,4 +37,24 @@ def call(params)
 
     super(params, {})
   end
+
+  private
+
+  def set_params_and_validate(params)
+    # Set patterns includes a data validation before assigning the value to the attribute.
+    # A failure should return a failure.
+    set_patterns(params) if params[:patterns].present?
+    return [false, type.errors] if type.errors.any?
+
+    super
+  end
+
+  def set_patterns(params)
+    Types::Patterns::Collection
+      .build(patterns: params[:patterns])
+      .either(
+        ->(collection) { type.patterns = collection },
+        ->(result) { type.errors.add(:patterns, result.errors(full: true).messages.join(", ")) }
```

Comment:
:green_circle: Sorry that I didn't see this one before. Shouldn't we loop and add all the errors, so that formatting (i.e. the `join(', ')`) is left as a choice to the final renderer?

I.e.
```suggestion
        ->(result) { result.errors(full: true).each { |e| type.errors.add(:patterns, e.message) }
```

Reply:
Solved in #17644